cloudbuild-e2e.yaml

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+# create a GKE cluster
```

Comment:
Do you have any ideas on ways to eliminate this step?  This probably makes a ~1 minute test a ~20 minute test.

Reply:
I think we actually could use 1 cluster, since this test is really just about testing the collector config. We could have 1 ClusterRole installed in the cluster and install each collector to its own test namespace. It will add a little complexity since we'll have to update the manifests with the generated namespace name, and ideally this test would drop the clusterrole manifest with kustomize so multiple PRs weren't stamping on it over each other. That could lead to weird confusing flakes.

I think that's something we should do, but if it's okay with you I'd like to make that a follow-up improvement so we can get this test merged. I'm ok with the extra time for now